tensorrt_llm/executor/rpc/rpc_client.py

```
@@ -267,31 +320,58 @@ def safe_set_exception(f=future, exc=exception):
-        return await self._client_socket.get_async()
+        # Use timeout-based recv to handle cancellation gracefully
+        # This prevents the CancelledError from being logged as an exception
+        while True:
+            try:
+                # Short timeout allows periodic checks for cancellation
+                return await self._client_socket.get_async_noblock(timeout=2)
+            except asyncio.TimeoutError:
+                # Check if we should exit due to cancellation
+                if self._closed or (self._reader_asyncio_task
+                                    and self._reader_asyncio_task.cancelled()):
+                    raise asyncio.CancelledError("Reader task cancelled")
```

Ruff (0.14.5)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)